### ISSUE TITLE
ExecuTorch filter subplugin & Tizen porting

### ISSUE THREAD

**Issue by myungjoo** [nnstreamer/nnstreamer#4289]
PyTorch is releasing ExecuTorch for edge device inferences.

- [ ] Tizen porting (tizen-next)
- [ ] tensor-fitler::ExecuTorch (in C++)
- [ ] test cases
- [ ] sample applications w/ GUI

**Comment by anyj0527** [comment]
REF:
https://github.com/pytorch/executorch
https://pytorch.org/executorch/stable/index.html

**Comment by leeeryboy** [comment]
Hi nnstreamer team,

My team(me, @jinotter3, @suzyvaque) is proposing the integration of the executorch tensor filter into nnstreamer. Supporting executorch could enhance nnstreamer's capabilities by enabling efficient execution of PyTorch models directly.

Currently, nnstreamer supports various tensor filters but lacks integration with executorch. Including this could not only expand nnstreamer's usability but also attract more PyTorch users to nnstreamer.

Thank you for considering this enhancement. I look forward to your feedback and any suggestions you might have.

**Comment by myungjoo** [comment]
1. Please write in a single C++ file, inheriting ```tensor_filter_subplugin``` class. For example, ```tensor_filter_snpe.cc```.
2. Please make it portable. It should be able to be built (and tested) in a new Ubuntu machine without any (unscripted) changes. Ultimately, it should be able to be built by pdebuild (or PPA) if you consider Ubuntu as the standard environment.
    - You may be based on the docker file (/tools/docker). If you can build executorch subplugin in the docker by adding one or two lines that "installs" executorch in the docker, you can ensure its portability.
    - Or, in other words, you need to guarantee that anyone else in the world can get the same build results without reading your build manual, special build scripts, or description.

**Comment by anyj0527** [comment]
FYI: There was a new release of ExecuTorch recently.
https://pytorch.org/blog/executorch-alpha/
https://github.com/pytorch/executorch/releases/tag/v0.2.0

**Comment by myungjoo** [comment]
Steps 1 and 2 of the plan in the status update above are in #4915, now out of draft with all checks green.

It turned out smaller than the plan assumed. ExecuTorch generates its own `executorch.pc` under `EXECUTORCH_BUILD_SHARED`, so meson needed no new detection logic — the work was a reproducible source-build recipe (`tools/executorch-install.sh`), one fixup to that generated file, and the CI job.

What the job proves, rather than asserts: the SSAT suite runs 8 of 8, two of those being byte-exact golden compares, so the operator kernels really are registered end to end.

Three things surfaced on the way that are worth recording against this EPIC:

- The generated `executorch.pc` links `libexecutorch.so` alone, which carries the runtime and extensions but **no operator kernels**. A subplugin built against the stock file links cleanly and fails at the first invoke. The kernels are in `libexecutorch_portable_ops.so` and are reached only through static registrars, so they need `--no-as-needed`.
- Building the kernels runs ExecuTorch's codegen, which needs `torchgen` — that is, the torch wheel. There is no usable standalone `torchgen` on PyPI, and ExecuTorch's own `Codegen.cmake` ignores the result of its torchgen lookup, so a missing one surfaces much later as a missing `tags.yaml`.
- One of the two 2024 `.pte` fixtures no longer loaded. `Program::load` takes the constant segment path only when `constant_segment.offsets` is non-empty, and v1.4.1 compiles out the inline `constant_buffer` fallback, so the fixture with no constants was refused with `InvalidProgram`. Both fixtures are regenerated and `generateModel.py` now rebuilds them.

Remaining on this EPIC, unchanged: the ExecuTorch 1.x API modernization (the subplugin is float32 only and still on the deprecated `torch::executor` aliases), the `executorch-llama` decision — #4915 stops it auto-enabling but does not revive it, and `llamacpp` already covers the LLM path — Tizen `executorch-devel`, and the sample applications.

**Comment by myungjoo** [comment]
#4915 is merged (`e861c738`). Two of the four checklist items are now done, and the two remaining pieces of subplugin work are split out so this EPIC can track them rather than carry them.

**Done**

- **tensor-filter::ExecuTorch (in C++)** — the code has been in the tree since 2024; it is now actually built.
- **test cases** — `unittest_filter_executorch` (12 cases) and the SSAT suite (8 cases) run on Ubuntu 22.04 via `.github/workflows/executorch.yml`, on PRs touching the relevant paths plus daily.

**Split out**

- #4922 — bring the subplugin up to the ExecuTorch 1.x API. The one with user-visible impact is that it accepts float32 only, which rules out every quantized `.pte`; the deprecated `torch::executor` namespace and the output `memcpy` ride along in the same three functions.
- #4923 — decide the fate of `executorch-llama`. It cannot build against 1.x (`example::Runner` is gone), #4915 only stopped it auto-enabling, and `llamacpp` now covers the LLM path. A maintainer's call, not a code task.

**Still open here**

- **Tizen porting (tizen-next)** — `packaging/nnstreamer.spec` keeps `executorch_support 0` in all four profiles. The `BuildRequires` and subpackage are already in place from #4647; this is waiting on an `executorch-devel` package on the platform, not on nnstreamer code.
- **sample applications w/ GUI** — not started, and it lives in `nnstreamer-example`.

Two things worth recording that are not checklist items but block real use:

- **`debian/` has no ExecuTorch at all**, so Ubuntu and PPA users still have to run `tools/executorch-install.sh` by hand. An `executorch` dev package in `ppa:nnstreamer` would ship the `.pc` and make that script unnecessary for them.
- **macOS is uncovered.** The build logic is written to work there, and `tools/executorch-install.sh` branches for the Mach-O linker, but no CI job exercises it. Deferred deliberately when #4915 was scoped Ubuntu-first.
